extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/BasicSecurityDruidModule.java

```diff
@@ -74,81 +78,149 @@ public void configure(Binder binder)
 
     Jerseys.addResource(binder, BasicAuthenticatorResource.class);
     Jerseys.addResource(binder, BasicAuthorizerResource.class);
+
+    binder.bind(MetadataStorage.class).toProvider(MetadataStorageProvider.class);
+    LifecycleModule.register(binder, MetadataStorage.class);
   }
 
-  @Provides @LazySingleton
-  public static BasicAuthenticatorMetadataStorageUpdater createAuthenticatorStorageUpdater(final Injector injector)
+  @Provides
+  @LazySingleton
+  public static BasicAuthenticatorMetadataStorageUpdater createAuthenticatorStorageUpdater(
+      final Injector injector,
+      BasicAuthClassCompositionConfig config
+  ) throws ClassNotFoundException
   {
+    if (config.getAuthenticatorMetadataStorageUpdater() != null) {
```

Comment:
This block of code can be extracted as a method.

Reply:
Please add some explanation. What is going on here?

Reply:
Refactored and added description